File: 0401-0500/0451-Sort Characters By Frequency/0451-Sort Characters By Frequency.py
import collections
class Solution:
    def frequencySort(self, s: str) -> str:
        table = collections.Counter(s)
        buckets = ['' for _ in range(len(s) + 1)]
        for k, v in table.items():
            buckets[v] += k*v
   
        return ''.join(buckets[::-1])

File: 0401-0500/0451-Sort Characters By Frequency/0451-Sort Characters By Frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> table;
        for (char c : s) {
            ++table[c];
        }
        
        priority_queue<pair<int, char>> pq;
        for (auto& p : table) {
            pq.push({p.second, p.first});
        }
        
        string result;
        while (!pq.empty()) {
            auto p = pq.top();
            pq.pop();
            result += string(p.first, p.second);
        }
        
        return result;
    }
};
